### PR TITLE
Doc hyperspy main branches and rebase in the dev guide

### PR DESCRIPTION
**Pull request by ericpre**
Clarify the differences between the main branches in the hyperspy repository and explain how to rebase.

### Progress of the PR
- [x] update pull request template,
- [x] update dev guide,
- [x] ready for review.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ### Requirements
 * Read the [developer guide](http://hyperspy.org/hyperspy-doc/current/dev_guide.html).
+* Base your pull request on the [correct branch](http://hyperspy.org/hyperspy-doc/current/dev_guide.html#semantic-versioning-and-hyperspy-main-branches).
```
Comment (thomasaarholt) [inline .github/PULL_REQUEST_TEMPLATE.md]: These two links (lines 2 and 3) don't work. I realise that you create the "semantic.." heading in this PR, but the `dev_guide.html` file doesn't exist in the built docs.

**Review by thomasaarholt — CHANGES_REQUESTED** [review]
This is a nice addition! The links in the md file are outdated though.

**Comment by ericpre** [comment]
Thanks for the review, the links are fixed.

**Review by thomasaarholt — APPROVED** [review]

**Comment by ericpre** [comment]
Thanks @thomasaarholt!

**Pull request merged**